ggml.c

```diff
@@ -16856,23 +16856,20 @@ static thread_ret_t ggml_graph_compute_thread(void * data) {
     return 0;
 }
 
-void ggml_graph_compute(struct ggml_context * ctx, struct ggml_cgraph * cgraph) {
-    const int n_threads = cgraph->n_threads;
+// Prepare for graph computing.
+// Will set: node->n_tasks, ctx->{work_size, planned}
+void ggml_graph_compute_plan(struct ggml_cgraph_context * ctx, struct ggml_cgraph * cgraph) {
```

Comment:
I saw https://github.com/ggerganov/ggml/issues/299#issuecomment-1616698106
Yes, it's better to return `ggml_cgraph_context` as @slaren suggested. 
```
ggml_cgraph_context plan = ggml_graph_compute_plan(&gf);
```